modules/user_management.js

```diff
@@ -130,6 +130,52 @@ class UserManagement {
         }]
       }
     })
+
+    // Add a recital ID (role name) to the record of all participants
+    dispatcher.command('record', this._guild.id, async (authorMember, tokenized) => {
+      util.requireRole(authorMember, this._managementRole)
+
+      // p!record @role
+      const USAGE = `${this._config.get('commandPrefix') || 'p!'}record @role`
+      util.requireParameterCount(tokenized, 1, USAGE)
+      util.requireParameterFormat(tokenized[0], arg => arg.startsWith('<@&') && arg.endsWith('>'), USAGE)
+
+      const recitalRole = await this._guild.roles.fetch(tokenized[0].replace(/[<@&>]/g, ''))
+
+      // Check that input role is a recital role
+      // Recital roles must be below config.recitalUpperBound and above config.recitalLowerBound
+      const upperBound = await this._guild.roles.fetch(this._config.get('recitalUpperBoundId'))
+      const lowerBound = await this._guild.roles.fetch(this._config.get('recitalLowerBoundId'))
+      if ((upperBound !== undefined && recitalRole.comparePositionTo(upperBound) >= 0) ||
+        (lowerBound !== undefined && recitalRole.comparePositionTo(lowerBound) <= 0)) {
+        throw new Error(`${tokenized[0]} is outside the range of valid recital roles.`)
+      }
+
+      // Set the name of recitalRole as the recital ID
+      const recitalId = recitalRole.name
+
+      // Some roles may in fact be better classified as server "Events" if they do not contain the word "Recital"
```

Comment:
Per offline discussion - I believe we were going to put everything in the same field?

Reply:
This is honestly something I'm really unsure about. Currently, this allows us to group Sonata Marathoner and Composer Festival 2021 separately, because they're not actually "recitals." They were pretty involved events that required a lot of work from organization and participation, and I was assuming that we would 100% want to treat these differently from the rest in the stats card -- hence the separate fields even if all the recitals (numbered or not) are consolidated.